ouroboros-network/src/Ouroboros/Network/Diffusion/P2P.hs

```diff
@@ -872,330 +866,326 @@ runM Interfaces
+                      $ \governorThread ->
+                        withSockets tracer diNtnSnocket
+                                    (\sock addr -> diNtnConfigureSocket sock (Just addr))
+                                    diNtnConfigureSystemdSocket
+                                    ( catMaybes
+                                        [ daIPv4Address
+                                        , daIPv6Address
+                                        ]
+                                    )
+                                    $ \sockets addresses -> do
+                        --
+                        -- Run server
+                        --
+                        traceWith tracer (RunServer addresses)
+                        Async.withAsync
+                          (Server.run
+                            ServerArguments {
+                                serverSockets               = sockets,
+                                serverSnocket               = diNtnSnocket,
+                                serverTracer                = dtServerTracer,
+                                serverTrTracer              = dtInboundGovernorTransitionTracer,
+                                serverInboundGovernorTracer = dtInboundGovernorTracer,
+                                serverConnectionLimits      = daAcceptedConnectionsLimit,
+                                serverConnectionManager     = connectionManager,
+                                serverInboundIdleTimeout    = Just daProtocolIdleTimeout,
+                                serverInboundInfoChannel    = inboundInfoChannel,
+                                serverObservableStateVar    = observableStateVar
+                              })
+                              $ \serverThread ->
+                                Async.withAsync
+                                  (Governor.peerChurnGovernor
+                                    dtTracePeerSelectionTracer
+                                    daDeadlineChurnInterval
+                                    daBulkChurnInterval
+                                    daPeerMetrics
+                                    churnModeVar
+                                    churnRng
+                                    daBlockFetchMode
+                                    daPeerSelectionTargets
+                                    peerSelectionTargetsVar)
+                                  $ \churnGovernorThread ->
+
+                                    -- wait for any thread to fail
+                                    snd <$> Async.waitAny
+                                         [ ledgerPeersThread
+                                         , localRootPeersProvider
+                                         , serverThread
+                                         , governorThread
+                                         , churnGovernorThread
+                                         ]
```

Comment:
This should make the diff cleaner.  There are other PRs in the queue which also touch this file, so it will make conflict resolution easier.
```
                                           [ localRootPeersProvider
                                           , serverThread
                                           , governorThread
                                           , ledgerPeerThread
                                           , churnGovernorThread
                                           ]
```